LocalHistory.py

```diff
@@ -306,3 +359,333 @@ def run(self, edit):
 
         shutil.rmtree(get_history_root())
         status_msg('The Local History has been deleted for all files.')
+
+class HistoryCreateSnapshot(sublime_plugin.TextCommand):
+
+    def on_done(self, string):
+        self.string = string
+        self.view.window().run_command('history_create_snapshot', {"callback": True})
+
+    def run(self, edit, callback=None):
+
+        if not callback:
+            v = self.view
+
+            file_name = os.path.basename(v.file_name())
+            self.pre, self.ext = os.path.splitext(file_name)
+            c = "Enter a name for this snapshot:  "
+            s = ""
+
+            v.window().show_input_panel(c, s, self.on_done, None, None)
+
+        else:
+            v = self.view
+            file_name = self.pre + " # " + self.string + self.ext
+            history_dir = get_history_subdir(v.file_name())
+            shutil.copyfile(v.file_name(), os.path.join(history_dir, file_name))
+            status_msg('File snapshot saved under "' + file_name + '".')
+
+class HistoryOpenSnapshot(sublime_plugin.TextCommand):
+
```

Comment:
Snapshots are saved in the same history folder, but they have no date and a # symbol instead, they look like this

![Imgur](https://i.imgur.com/XOjaqOZ.png)